core/tests/protractor_desktop/voiceoverUploadFeatures.js

```diff
@@ -85,6 +89,49 @@ describe('Voiceover upload features', function() {
       '../data/cafe-over-five-minutes.mp3');
     explorationEditorTranslationTab.expectSaveUploadedAudioButtonToBeDisabled();
     explorationEditorTranslationTab.closeUploadAudioModal();
+    explorationEditorTranslationTab.deleteAudioRecord();
+  });
+
+  it('should upload recorded audio', function() {
+    explorationEditorTranslationTab.addAudioRecord();
+    explorationEditorTranslationTab.stopAudioRecord();
+    explorationEditorTranslationTab.confirmAudioRecord();
+    explorationEditorTranslationTab.playAudioRecord();
+    browser.refresh();
+    explorationEditorTranslationTab.playAudioRecord();
+  });
+
+  it('should play recorded file after logging out', function() {
+    users.logout();
+    users.login(TEST_EMAIL);
+    creatorDashboardPage.get();
+    creatorDashboardPage.editExploration('Untitled');
+    explorationEditorMainTab.exitTutorial();
+
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.playAudioRecord();
+    explorationEditorTranslationTab.deleteAudioRecord();
+  });
+
+  it('should upload audio file from path', function() {
+    explorationEditorTranslationTab.uploadAudioRecord(
+      '../../../data/explorations/audio_test/assets/audio/test_audio_1_en.mp3');
+    explorationEditorTranslationTab.saveAudioRecord();
+    explorationEditorTranslationTab.playAudioRecord();
+    browser.refresh();
+    explorationEditorTranslationTab.playAudioRecord();
+  });
+
+  it('should play recorded file from file after logging out', function() {
```

Comment:
Ditto here. I think for both of these, you can club the 2 it blocks and have just one test for both logged in and logged out flow.

Reply:
Resolved by 70592c2e6